File: app/src/main/java/com/bojie/weatherBo/adapter/HourAdapter.java
package com.bojie.weatherbo.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bojie.weatherbo.R;
import com.bojie.weatherbo.ui.MainActivity;
import com.bojie.weatherbo.util.UnitConvert;
import com.bojie.weatherbo.weather.Hour;

import java.text.DecimalFormat;

/**
 * Created by bojiejiang on 3/5/15.
 */
public class HourAdapter extends RecyclerView.Adapter<HourAdapter.HourViewHolder> {

    private Hour[] mHours;
    private Context mContext;
    double mTransferredTemp;

    public HourAdapter(Context contexts, Hour[] hours) {
        mHours = hours;
        mContext = contexts;
    }


    @Override
    public HourViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.hourly_list_item, parent, false);
        HourViewHolder viewHolder = new HourViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(HourViewHolder holder, int position) {
        holder.bindHour(mHours[position]);
    }

    @Override
    public int getItemCount() {
        return mHours.length;
    }

    public class HourViewHolder extends RecyclerView.ViewHolder
            implements View.OnClickListener {

        public TextView mTimeLabel;
        public TextView mSummaryLabel;
        public TextView mTemperatureLabel;
        public ImageView mIconImageView;


        public HourViewHolder(View itemView) {
            super(itemView);
            mTimeLabel = (TextView) itemView.findViewById(R.id.tv_hourlyTimeLabel);
            mSummaryLabel = (TextView) itemView.findViewById(R.id.tv_hourlySummaryLabel);
            mTemperatureLabel = (TextView) itemView.findViewById(R.id.tv_hourlyTemperatureLabel);
            mIconImageView = (ImageView) itemView.findViewById(R.id.hourlyIconImageView);

            itemView.setOnClickListener(this);
        }

        public void bindHour(Hour hour) {
            mTimeLabel.setText(hour.getHour());
            mSummaryLabel.setText(hour.getSummary());
            // Convert unit
            double TempInF = hour.getTemperature();
            double TempInC = UnitConvert.fahrenheitToCelsius(TempInF);
            mTemperatureLabel.setText(Math.round(TempInF) + "");
            mTransferredTemp = TempInF;
            if (MainActivity.mButtonUnitConvert.getText() == "F") {
                mTransferredTemp = TempInF;
                mTemperatureLabel.setText(Math.round(TempInF) + "");
            } else if (MainActivity.mButtonUnitConvert.getText() == "C") {
                mTransferredTemp = TempInC;
                mTemperatureLabel.setText(Math.round(TempInC) + "");
            }
            mIconImageView.setImageResource(hour.getIconId());
        }

        @Override
        public void onClick(View v) {
            String time = mTimeLabel.getText().toString();
            // Format decimal
            DecimalFormat df = new DecimalFormat();
            df.setMaximumFractionDigits(2);

            String temperature = df.format(mTransferredTemp);
            Log.d("temp",temperature);
            String summary = mSummaryLabel.getText().toString();
            String message = String.format("At %s it will be %s and %s",
                    time,
                    temperature,
                    summary);

            Toast.makeText(mContext, message, Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/com/bojie/weatherBo/adapter/HourAdapter.kt
package com.bojie.weatherbo.adapter

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.bojie.weatherbo.R
import com.bojie.weatherbo.ui.MainActivity
import com.bojie.weatherbo.util.UnitConvert
import com.bojie.weatherbo.weather.Hour
import java.text.DecimalFormat
import kotlin.math.round

class HourAdapter(private val mContext: Context, private val mHours: Array<Hour>) : 
    RecyclerView.Adapter<HourAdapter.HourViewHolder>() {
    
    private var mTransferredTemp: Double = 0.0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HourViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.hourly_list_item, parent, false)
        return HourViewHolder(view)
    }

    override fun onBindViewHolder(holder: HourViewHolder, position: Int) {
        holder.bindHour(mHours[position])
    }

    override fun getItemCount(): Int {
        return mHours.size
    }

    inner class HourViewHolder(itemView: View) : 
        RecyclerView.ViewHolder(itemView), View.OnClickListener {
        
        val timeLabel: TextView = itemView.findViewById(R.id.tv_hourlyTimeLabel)
        val summaryLabel: TextView = itemView.findViewById(R.id.tv_hourlySummaryLabel)
        val temperatureLabel: TextView = itemView.findViewById(R.id.tv_hourlyTemperatureLabel)
        val iconImageView: ImageView = itemView.findViewById(R.id.hourlyIconImageView)

        init {
            itemView.setOnClickListener(this)
        }

        fun bindHour(hour: Hour) {
            timeLabel.text = hour.hour
            summaryLabel.text = hour.summary
            
            // Convert unit
            val tempInF = hour.temperature
            val tempInC = UnitConvert.fahrenheitToCelsius(tempInF)
            
            mTransferredTemp = tempInF
            
            when (MainActivity.mButtonUnitConvert.text) {
                "F" -> {
                    mTransferredTemp = tempInF
                    temperatureLabel.text = round(tempInF).toInt().toString()
                }
                "C" -> {
                    mTransferredTemp = tempInC
                    temperatureLabel.text = round(tempInC).toInt().toString()
                }
                else -> {
                    temperatureLabel.text = round(tempInF).toInt().toString()
                }
            }
            
            iconImageView.setImageResource(hour.iconId)
        }

        override fun onClick(v: View) {
            val time = timeLabel.text.toString()
            
            // Format decimal
            val df = DecimalFormat().apply {
                maximumFractionDigits = 2
            }

            val temperature = df.format(mTransferredTemp)
            Log.d("temp", temperature)
            val summary = summaryLabel.text.toString()
            val message = String.format("At %s it will be %s and %s",
                time,
                temperature,
                summary)

            Toast.makeText(mContext, message, Toast.LENGTH_SHORT).show()
        }
    }
} 